sdk/keyvault/keyvault-keys/test/CRUD.test.ts

```diff
@@ -271,7 +274,10 @@ describe("Keys client - create, read, update and delete operations", () => {
       throw Error("Expecting an error but not catching one.");
     } catch (e) {
       if (e instanceof RestError) {
-        assert.equal(e.message, `Key not found: ${keyName}`);
+        assert.equal(
+          e.message.split(".")[0],
+          `A key with (name/id) ${keyName} was not found in this key vault`
```

Comment:
If we search for the string "not found", both the old and new would match. I wonder if we should just try to be more flexible here instead of moving to a new message

Reply:
Can we test the status code or error code instead of the error message?

Reply:
Both sound good! I'll try with the status code first.

Though it does concern me that this happened. I'll try to get an answer of why did this happen this week, just so that we're able to know more about these changes preemptively.

Reply:
I just pushed the change! I also reverted the recording changes, since this does work with both the older API shape and the newer one. I checked this by running the playback tests against the new recordings, and then against the older ones.